### DISCUSSION TITLE
How do I convert floating (undefined) states to "low" but still maintaining compatibility with FPGAs?

### DISCUSSION THREAD

**Discussion by Slicudis** [logisim-evolution/logisim-evolution#2010] [Q&A]
I'm designing a CPU for my FPGA, but undefined states are making this harder. I can't use pulldown resistors, because they're uncompatible with the FPGA synthetizer. Could anyone please send me a schematic if you have the solution to that?

**Comment by maehne** [comment]
Your question is too generic. You need to provide more context and ideally a small reproducer demonstrating what you would like to achieve to increase your chance for getting a good answer. In general, we recommend our students to strive for a fully synchronous design with one clock and either an asynchronous[^1] or synchronous[^2] reset for putting your system in a well-defined state after power up. In case you want to implement a bus for communicating with some peripherals, you should typically use separate busses for the data put by the CPU (initiator) on the bus and the data coming from the targets to the bus. If all targets drive the bus to low when inactive, the data signals can be simply combined with a big OR gate that has as many inputs as connected targets.

[^1]: using the asynchronous reset and set inputs of flip-flops/registers
[^2]: reset signal part of next-state logic, often implemented as a single two-to-one Mux right in front of register to inject the initial state value
